src/tasks/deployment-of-bridged-token-deployer.ts

```diff
@@ -0,0 +1,135 @@
+  task(
+    "deployment-bridged-token-deployer",
+    "Generate the list of claims from a csv and reads settings from json and deploys the bridged token deployer on gnosis chain",
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .setAction(generateDeployment);
+};
+
+async function generateDeployment(
+  { claims: claimCsv, settings: settingsJson }: Args,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> {
+  const { ethers } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const chainId = (await hre.ethers.provider.getNetwork()).chainId.toString();
+  if (chainId !== "100") {
+    throw new Error(
+      `This script must be run on gnosis chain. Found chainId ${chainId}`,
+    );
+  }
+
+  const inputSettings: Settings = JSON.parse(
+    await fs.readFile(settingsJson, "utf8"),
+  );
+  console.log(`Using deployer ${deployer.address}`);
+
+  console.log("Reading user claims for gnosis chain from file...");
+  const claims = await parseCsvFile(claimCsv);
+
+  console.log("Generating Merkle proofs...");
+  const { merkleRoot, claims: claimsWithProof } = computeProofs(claims);
+
+  const settings = {
+    ...inputSettings,
+    virtualCowToken: {
+      gnoPrice: "0",
+      nativeTokenPrice: "0",
+      merkleRoot: constants.HashZero,
+      usdcToken: constants.AddressZero,
+      gnoToken: constants.AddressZero,
+      wrappedNativeToken: constants.AddressZero,
+    },
+    multiTokenMediatorGnosisChain: constants.AddressZero,
+  };
+
+  // In the following function, we are generating the addresses, as they would
+  // be generated within the mainnet deployment script - but with many zero
+  // addresses and hashes, as displayed in the settings definition
+  // Hence, its very important generateProposal generates the same address
+  // for the cowDao and cowToken anyways.
+  // We have the unit test: invariance of cowDao and cowToken that checks that.
+  const { addresses } = await generateProposal(
+    settings,
+    defaultSafeDeploymentAddresses(chainId),
+    hre.ethers,
+  );
+
+  const deploymentHelperParameters: DeploymentHelperDeployParams = {
+    foreignToken: addresses.cowToken,
+    multiTokenMediatorGnosisChain:
+      settings.bridge.multiTokenMediatorGnosisChain,
+    merkleRoot,
+    communityFundsTarget: addresses.cowDao,
+    gnoToken: defaultTokens.gno[chainId],
+    gnoPrice: settings.virtualCowToken.gnoPrice,
+    nativeTokenPrice: nativeTokenPriceGnosisChain,
+    wrappedNativeToken: defaultTokens.weth[chainId],
+  };
+
+  const BridgedTokenDeployer = await hre.ethers.getContractFactory(
+    "BridgedTokenDeployer",
+  );
+  const bridgedTokenDeployer = await BridgedTokenDeployer.deploy(
+    ...constructorInput(
+      ContractName.BridgedTokenDeployer,
+      deploymentHelperParameters,
+    ),
+  );
```

Comment:
I still think this could go to its own function, so that we (eventually) can test it properly. E.g., are the settings respected in the deployed factory? Then we could also more easily test combining the two function together. We run this, then we run `generateProposal` with the updated settings from this. Are the addresses the same?

Note that this is the core logic of this script, so once we test this we could be decently confident that everything works. I point this out because the deployment script was [built with this idea in mind](https://github.com/gnosis/cow-token/blob/main/src/tasks/deployment.ts#L76-L90), and I personally find tests like [these](https://github.com/gnosis/cow-token/blob/main/test/proposal.test.ts#L102-L106) very reassuring.

Reply:
I don't know how to run all together in one script. Because first one needs to be connected to gnosis chain and then to Ethereum chain. Not sure how easy or confusing it is to switch the provider in-between.

Maybe we can leave that for later?

Reply:
Mmh, true. But in principle we could cheat a bit and deploy everything in the same network, I'd argue that it would be a decent approximation.
Not a thing for this PR anyway!

Reply:
Also I think it's doable to test simultaneously in both networks. [This](https://github.com/gnosis/cow-token/blob/main/test/mainnet/chain-fork.ts) is how mainnet is handled, changing the URL is the easy part.